File: 0189-rotate-array/0189-rotate-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    let res=[]
    let n=nums.length-k
    
    while(n<0){
        n+=nums.length
    }
    for(let i=n;i<nums.length;i++){
        res.push(nums[i])
    }
    // console.log(res)
    let diff=nums.length-k
    while(diff<0){
        diff+=nums.length
    }
    for(let j=0;j<diff;j++){
        res.push(nums[j])
    }
    for(let i=0;i<res.length;i++){
        nums[i]=res[i]
    }
};

File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        dif=k%len(nums)
        temp1=nums[:len(nums)-dif]
        temp2=nums[len(nums)-dif:]
        for i in range(len(temp2)):
            nums[i]=temp2[i]
        for i in range(len(temp1)):
            nums[i+dif]=temp1[i]